packages/flutter_tools/lib/src/build_system/targets/icon_tree_shaker.dart

```diff
@@ -150,15 +156,24 @@ class IconTreeShaker {
   /// If the font-subset subprocess fails, it will [throwToolExit].
   /// Otherwise, it will return true.
   Future<bool> subsetFont({
-    @required String inputPath,
+    @required File input,
     @required String outputPath,
     @required String relativePath,
   }) async {
     if (!enabled) {
       return false;
     }
-
-    _iconData ??= await _getIconData(_environment);
+    if (input.lengthSync() < 12) {
+      return false;
+    }
+    final String mimeType = mime.lookupMimeType(
```

Comment:
How confident are you that this won't cause new problems?

Reply:
I'm not confident about anything

Reply:
Maybe we can just move the mime chekc to a separate pr

Reply:
We might as well resolve this now if you have concerns :) 

Are there non-ttf fonts that the icon_tree_shaker works with? Or does the hb subset utility have a list of supported formats?

Reply:
I guess this should be fairly safe.  Mime sniffing can get funny, and we could get broken if the mime package changes the spelling of a type

Reply:
I'll add some more tests around this logic

Reply:
More tests added. If the spelling changes there was an existing test that fails, but I also added cases to verify we don't subset invalid fonts or non-ttf mime types